packages/terra-search-field/src/SearchField.jsx

```diff
@@ -107,15 +111,15 @@ class SearchField extends React.Component {
     this.handleTextChange = this.handleTextChange.bind(this);
     this.handleSearch = this.handleSearch.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleInput = this.handleInput.bind(this);
     this.setInputRef = this.setInputRef.bind(this);
-    this.updateSearchText = this.updateSearchText.bind(this);
 
     this.searchTimeout = null;
-    this.state = { searchText: this.props.defaultValue || this.props.value };
+    this.searchText = this.props.defaultValue || this.props.value;
```

Comment:
Why did we remove this from state?

Reply:
Since we now have both a handleInput and handleTextChange, when you type a value into the input they both fire. Since setState is asynchronous this meant most of the time both would run the updateSearchText forceUpdate calls.

Reply:
> Since setState is asynchronous this meant most of the time both would run the updateSearchText forceUpdate calls.

what do you mean?

Reply:
Typing the value into the components results in onInput and onChange being called in quick succession of each other. Fast enough that when I left it as a state and put in console messages to follow logic flow, I kept getting multiple setState calls on each keystroke.